src/node/nodestate.h

```diff
@@ -402,9 +402,19 @@ namespace ccf
 
             self = resp.node_id;
 
-            setup_consensus(args.consensus_type, args.config, resp.public_only);
+            if (resp.consensus_type != network.consensus_type)
```

Comment:
Alternative proposal that I think minimises the number of arguments passed to `cchost` and reduces the probability of operators passing the wrong arguments to `cchost`: only the first node (i.e. `cchost start`) is given the consensus type while joining nodes (i.e. `cchost join`) are not passed the consensus type directly but retrieve it from the primary node on join. That way, there is no need for this check.

Reply:
We need to pass the consensus to the Enclave on node startup because that is where the `NetworkState` object is created which in turn creates the `NetworkTables` object which creates the tables, and we need to consensus at that point so that we know if the tables should be derived or replicated